_aggregations/pipeline/moving_fn.md

```diff
@@ -0,0 +1,611 @@
+
+
+## How moving function works
+
+The `moving_fn` aggregation operates on a sliding window over an ordered sequence of buckets. Starting at the first bucket in the parent aggregation, `moving_fn` does the following:
+
+1. Collects the subsequence (window) of values from the buckets specified by the `window` and `shift` parameters.
+2. Passes these values as an array to the function specified by `script`.
+3. Uses `script` to compute a single value from the array.
+4. Returns this value as the result for the current bucket.
+5. Moves forward one bucket and repeats this process.
+
+"Past" and "future" values imply time-series data, the most common use case for moving window functions. More generally, you can take them to refer to previous and upcoming values, respectively, in any ordered data sequence.
+{: .note}
+
+The script applied by `moving_fn` can be a [predefined function](#predefined-functions) or a [custom script](#custom-scripts). Bucket values are provided to the script in the `values` array. The script returns a double value as the result. The result values `NaN` and `+/- Inf` are allowed, but `null` is not.
+
+
+### Window size
+
+The `window` parmeter specifies the number of buckets defining the size of the window.
+
+The array passed to the `script` function is zero-indexed. Its values are accessed within the script as `values[0]` to `values[n]`, where `n = values.length - 1`.
+
+
+### Shifting the window
+
+The `shift` parameter controls where the moving window lies relative to the current bucket. Set `shift` based on whether your analysis requires historical context, current data, or future prediction. The default is `0`, which shows only past values (excluding the current bucket). 
+
+Some commonly used values of `shift` are as follows:
+
+| `shift` | Window description                            |                     |
+| :--           | :--                                           | :--                 |
+| `0`           | Only past values. Excludes the current value. | `--[-----]x----`    |
+| `1`           | Past values, including the current value.     | `--[----x]-----`    |
+| `window/2`    | Centers the window around the current value.  | `--[--x--]-----`    |
+| `window`      | Future values, including the current value.   | `--[x----]-----`    |
+
+When a window extends beyond available data at the beginning or end of a sequence, `window` shrinks automatically to use only the available points:
+
+```
+[x----]--
+-[x----]-
+--[x----]
+---[x---]
+----[x--]
+-----[x-]
+------[x]
+```
+
+
+## Predefined functions
+
+The `moving_fn` aggregation supports a number of predefined functions that can be used instead of a custom script. The functions are accessible from the `MovingFunctions` context. For example, access the `max` function as `MovingFunctions.max(values)`. 
```

Comment:
```suggestion
The `moving_fn` aggregation supports a number of predefined functions that can be used instead of a custom script. The functions are accessible from the `MovingFunctions` context. For example, you can access the `max` function as `MovingFunctions.max(values)`. 
```